xonsh/jobs.py

```diff
@@ -364,7 +422,7 @@ def format_job_string(num: int, format="dict") -> str:
         "cmd": " ".join(
             [" ".join(i) if isinstance(i, list) else i for i in job["cmds"]]
         ),
-        "pid": int(job["pids"][-1]) if job["pids"] else None,
+        "pids": job["pids"] if "pids" in job else None,
     }
 
     if format == "posix":
```

Comment:
If someone passes `jobs --posix`, the code below still refers to the `pid` key but it should be `pids`